lib/buffer.js

```diff
@@ -1277,13 +1279,46 @@ function atob(input) {
       // The first 5 elements of `kForgivingBase64AllowedChars` are
       // ASCII whitespace char codes.
       nonAsciiWhitespaceCharCount++;
+
+      if (index === kForgivingBase64AllowedChars.length - 1 && equalCharCount !== 2) {
+        // The last element of `kForgivingBase64AllowedChars` is the `=`
+        equalCharCount++;
+      } else if (equalCharCount) {
+        // Only one more `=` is permitted after the first equal sign.
+        throw lazyDOMException('Invalid character', 'InvalidCharacterError');
+      }
     } else if (index === -1) {
       throw lazyDOMException('Invalid character', 'InvalidCharacterError');
     }
   }
 
+  // See #2 and #4 - https://infra.spec.whatwg.org/#forgiving-base64
+  if (equalCharCount > 2) {
+    throw lazyDOMException('Invalid character', 'InvalidCharacterError');
+  }
+
+  let reminder = nonAsciiWhitespaceCharCount % 4;
+
+  // See #2 - https://infra.spec.whatwg.org/#forgiving-base64
+  if (!reminder) {
+    // Remove all trailing `=` characters (whitespace characters will be
+    // ignored).
+    input = input.replace(/(=\s*){1,2}$/, (s) => {
+      return s.replace(/=/gi, '');
+    });
+    const deltaLength = initLength - input.length;
+    nonAsciiWhitespaceCharCount -= deltaLength;
+    equalCharCount -= deltaLength;
+    reminder = nonAsciiWhitespaceCharCount % 4;
+  }
+
+  // See #4 - https://infra.spec.whatwg.org/#forgiving-base64
+  if (equalCharCount) {
+    throw lazyDOMException('Invalid character', 'InvalidCharacterError');
+  }
```

Comment:
I don't think we need any of that, do we?
```suggestion
```

Reply:
I've updated the logic